search-app/src/cmr/search/results_handlers/results_handler_util.clj

```diff
@@ -0,0 +1,59 @@
+(ns cmr.search.results-handlers.results-handler-util
+  "Provides useful functions for several of the result handlers."
+  (:require
+   [clojure.set :as s]
+   [cmr.common.util :as util]))
+
+(defn build-each-concept-association-list
+  "This function builds a map with the passed in conept-key as the key and
+  a list of concept ids from the passed in associations."
+  [concept-key associations for-concept-type]
+  (if (= :collection for-concept-type)
+    {concept-key (or (seq (util/remove-nils-empty-maps-seqs
+                           (mapv :variable-concept-id associations)))
+                     (seq (util/remove-nils-empty-maps-seqs
+                           (mapv :service-concept-id associations)))
+                     (seq (util/remove-nils-empty-maps-seqs
+                           (mapv :tool-concept-id associations)))
+                     associations)}
+    {concept-key (or (seq (util/remove-nils-empty-maps-seqs
+                           (mapv :associated-concept-id associations)))
+                     associations)}))
+
+(defn build-association-concept-id-list
+  "Builds the association list from the passed in associations."
+  [associations for-concept-type]
+  (into {}
+        (map #(build-each-concept-association-list % (get associations %) for-concept-type)
+             (keys associations))))
+
+(defn build-detail-assoc-struct
+  "Builds the detailed association structure."
+  [association for-concept-type]
+  (if (= java.lang.String (type association))
```

Comment:
As a simplification could this be just `(string? association)` or would that be true for something else not intended?

Reply:
Fixed.